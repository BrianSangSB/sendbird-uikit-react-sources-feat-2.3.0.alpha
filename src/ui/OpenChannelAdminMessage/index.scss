@import '../../styles/variables';

.sendbird-openchannel-admin-message {
  position: relative;
  margin: 16px 24px;
  padding: 12px 16px;
  border-radius: 4px;
  @include themed() {
    background-color: t(bg-1);
  }
  &:hover {
    @include themed() {
      background-color: t(bg--1-4);
    }
  }

  .sendbird-openchannel-admin-message__text {
    // word-break: break-all;
    word-wrap: break-word;
  }
}
