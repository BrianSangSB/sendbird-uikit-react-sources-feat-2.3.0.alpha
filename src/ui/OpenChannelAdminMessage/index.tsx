import React, { ReactElement } from 'react';
import { ClientAdminMessage } from '../../index';
import './index.scss';

import Label, { LabelColors, LabelTypography } from '../Label';

interface Props {
  className?: string | Array<string>;
  message: ClientAdminMessage;
}

export default function OpenChannelAdminMessage({
  className,
  message,
}: Props): ReactElement {
  return (
    <div
      className={[
        ...(Array.isArray(className) ? className : [className]),
        'sendbird-openchannel-admin-message',
      ].join(' ')}
    >
      <Label
        className="sendbird-openchannel-admin-message__text"
        type={LabelTypography.CAPTION_2}
        color={LabelColors.ONBACKGROUND_2}
      >
        {message.message || ''}
      </Label>
    </div>
  );
}
