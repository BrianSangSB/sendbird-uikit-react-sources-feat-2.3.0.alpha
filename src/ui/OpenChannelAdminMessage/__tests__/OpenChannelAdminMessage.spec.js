import React from 'react';
import renderer from 'react-test-renderer';

import OpenChannelAdminMessage from "../index";

describe('OpenChannelAdminMessage', () => {
  it('should do a snapshot test of the OpenChannelAdminMessage DOM', function() {
    const component = renderer.create(
      <OpenChannelAdminMessage
        message={{ message: 'Hello I am Admin' }}
      />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
