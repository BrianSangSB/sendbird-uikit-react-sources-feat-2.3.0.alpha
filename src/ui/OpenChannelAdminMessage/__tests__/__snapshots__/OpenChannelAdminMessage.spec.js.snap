// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OpenChannelAdminMessage should do a snapshot test of the OpenChannelAdminMessage DOM 1`] = `
<div
  className=" sendbird-openchannel-admin-message"
>
  <span
    className="sendbird-openchannel-admin-message__text sendbird-label sendbird-label--caption-2 sendbird-label--color-onbackground-2"
  >
    Hello I am Admin
  </span>
</div>
`;
